feature/experimental/qos/otg_tests/two_sp_queue_traffic_test/two_sp_queue_traffic_test.go

```diff
@@ -0,0 +1,1586 @@
+				t.Logf("Get flow %q: lossPct: %.2f%% or rxPct: %.2f%%, want: %.2f%%\n\n", data.queue, lossPct, 100.0-lossPct, data.expectedThroughputPct)
+				if got, want := 100.0-lossPct, data.expectedThroughputPct; got < want-tolerance || got > want+tolerance {
+					t.Errorf("Get(throughput for queue %q): got %.2f%%, want within [%.2f%%, %.2f%%]", data.queue, got, want-tolerance, want+tolerance)
+				}
+			}
+
+			// Check QoS egress packet counters are updated correctly.
+			t.Logf("QoS dutQosPktsBeforeTraffic: %v", dutQosPktsBeforeTraffic)
+			t.Logf("QoS dutQosPktsAfterTraffic: %v", dutQosPktsAfterTraffic)
+			t.Logf("QoS dutQosDroppedPktsBeforeTraffic: %v", dutQosDroppedPktsBeforeTraffic)
+			t.Logf("QoS dutQosDroppedPktsAfterTraffic: %v", dutQosDroppedPktsAfterTraffic)
+			t.Logf("QoS ateOutPkts: %v", ateOutPkts)
+			t.Logf("QoS ateInPkts: %v", ateInPkts)
+			for _, data := range trafficFlows {
+				qosCounterDiff := dutQosPktsAfterTraffic[data.queue] - dutQosPktsBeforeTraffic[data.queue]
+				ateCounterDiff := ateInPkts[data.queue]
+				ateDropCounterDiff := ateOutPkts[data.queue] - ateInPkts[data.queue]
+				dutDropCounterDiff := dutQosDroppedPktsAfterTraffic[data.queue] - dutQosDroppedPktsBeforeTraffic[data.queue]
+				t.Logf("QoS queue %q: ateDropCounterDiff: %v dutDropCounterDiff: %v", data.queue, ateDropCounterDiff, dutDropCounterDiff)
+				if qosCounterDiff < ateCounterDiff {
+					t.Errorf("Get telemetry packet update for queue %q: got %v, want >= %v", data.queue, qosCounterDiff, ateCounterDiff)
+				}
+			}
+		})
+	}
+}
+
+func ConfigureDUTIntf(t *testing.T, dut *ondatra.DUTDevice) {
+	t.Helper()
+	dp1 := dut.Port(t, "port1")
+	dp2 := dut.Port(t, "port2")
+	dp3 := dut.Port(t, "port3")
+
+	dutIntfs := []struct {
+		desc      string
+		intfName  string
+		ipAddr    string
+		prefixLen uint8
+	}{{
+		desc:      "Input interface port1",
+		intfName:  dp1.Name(),
+		ipAddr:    "198.51.100.0",
+		prefixLen: 31,
+	}, {
+		desc:      "Input interface port2",
+		intfName:  dp2.Name(),
+		ipAddr:    "198.51.100.2",
+		prefixLen: 31,
+	}, {
+		desc:      "Output interface port3",
+		intfName:  dp3.Name(),
+		ipAddr:    "198.51.100.4",
+		prefixLen: 31,
+	}}
+
+	// Configure the interfaces.
+	for _, intf := range dutIntfs {
+		t.Logf("Configure DUT interface %s with attributes %v", intf.intfName, intf)
+		i := &oc.Interface{
+			Name:        ygot.String(intf.intfName),
+			Description: ygot.String(intf.desc),
+			Type:        oc.IETFInterfaces_InterfaceType_ethernetCsmacd,
+			Enabled:     ygot.Bool(true),
+		}
+		i.GetOrCreateEthernet()
+		s := i.GetOrCreateSubinterface(0).GetOrCreateIpv4()
+		if *deviations.InterfaceEnabled && !*deviations.IPv4MissingEnabled {
+			s.Enabled = ygot.Bool(true)
+		}
+		a := s.GetOrCreateAddress(intf.ipAddr)
+		a.PrefixLength = ygot.Uint8(intf.prefixLen)
+		gnmi.Replace(t, dut, gnmi.OC().Interface(intf.intfName).Config(), i)
+	}
+}
+
+func ConfigureQoS(t *testing.T, dut *ondatra.DUTDevice) {
```

Comment:
There have been a fair number of changes with the ATE test for ConfigureQos. You may just want to recopy ConfigureQoS and ConfigureCiscoQos from that test.

Reply:
Updated the OTG test to be aligned with changes in ATE